io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -299,15 +320,38 @@ impl<'c> NexusChild<'c> {
         })?;
         self.device_descriptor = Some(desc);
 
-        self.set_state(opened_state);
+        self.set_state(ChildState::Open);
+        self.sync_state = sync_state;
 
         info!("{:?}: opened successfully", self);
         Ok(self.name.clone())
     }
 
-    /// Check if we're healthy.
-    pub(crate) fn is_healthy(&self) -> bool {
+    /// Returns the state of the child.
+    pub fn state(&self) -> ChildState {
+        self.state.load()
+    }
+
+    /// Returns the sync state of the child.
+    pub fn sync_state(&self) -> ChildSyncState {
+        self.sync_state
+    }
+
+    /// Determines if the child is opened but out-of-sync.
```

Comment:
Ah I had missed this, it's already implied here, wonder if it's better not to be implied? 
Maybe it's clearer to have 2 functions, `is_out_of_sync` and `is_open_out_of_sync` ?

Reply:
We don't really need a non-Open out-of-sync.